addons/web/models/models.py

```diff
@@ -628,7 +628,7 @@ def formatter_many2many(value):
 
             return formatter_many2many
 
-        if field.type == 'many2one':
+        if field.type == 'many2one' or field_name == 'id':
```

Comment:
This is ok, note that `_read_group_postprocess_groupby` has a condition `field.relational or field.name == 'id'`. So I would expect this to be handled before many2many and then just `field.relational` that included many2many, one2many, etc.